### ISSUE TITLE
df.plot legend entries missing markers when plotting multiple dataframes on same axes

### ISSUE THREAD

**Issue by shahvi** [pandas-dev/pandas#28451]
#### Code Sample, a copy-pastable example if possible

```python
import pandas as pd
import matplotlib.pyplot as plt
df1 = pd.DataFrame(data=np.random.rand(100,1))
df1 = pd.DataFrame(data=np.random.rand(100,1))
a = plt.subplot()
df1.plot(ax=a,style='.')
df2.plot(ax=a,style='.')
```
#### Problem description
When plotting multiple dataframe on the same axes AND using the style '.' , only the last legend entry marker appears, the others remain blank.
Note, this problem doesn't appear when using the default plotting style.

#### Expected Output
All the legend entries should correctly match the various plots.

#### Output of ``pd.show_versions()``
<details>
[paste the output of ``pd.show_versions()`` here below this line]
INSTALLED VERSIONS
------------------
commit: None
python: 2.7.16.final.0
python-bits: 64
OS: Darwin
OS-release: 18.7.0
machine: x86_64
processor: i386
byteorder: little
LC_ALL: en_US.UTF-8
LANG: en_US.UTF-8
LOCALE: en_US.UTF-8

pandas: 0.24.2
pytest: None
pip: 19.2.2
setuptools: 41.0.1
Cython: None
numpy: 1.16.4
scipy: None
pyarrow: None
xarray: None
IPython: 5.8.0
sphinx: 1.8.5
patsy: None
dateutil: 2.8.0
pytz: 2019.2
blosc: None
bottleneck: None
tables: None
numexpr: None
feather: None
matplotlib: 2.2.3
openpyxl: None
xlrd: None
xlwt: None
xlsxwriter: None
lxml.etree: None
bs4: None
html5lib: None
sqlalchemy: None
pymysql: None
psycopg2: None
jinja2: 2.10.1
s3fs: None
fastparquet: None
pandas_gbq: None
pandas_datareader: None
gcsfs: None
</details>

**Comment by darkthunder007** [comment]
i tried the example,it wo
<img width="502" alt="df" src="https://user-images.githubusercontent.com/53977169/64914171-b11ee380-d75e-11e9-9f68-1a3067b18a8d.png">
rks

**Comment by jimmybeckett** [comment]
It worked for me too.

![image](https://user-images.githubusercontent.com/29718592/64918522-4799d780-d754-11e9-8125-2fec78cae17b.png)

Could you upload a screenshot of your plot so we can see exactly what looks wrong?

**Comment by darkthunder007** [comment]
Can you post the code, so that we can see what's not working

**Comment by shahvi** [comment]
@darkthunder007 I don't think the legend showed up correctly in your plot.  You are seeing the point behind the legend in your image.  I have added a plot showing the problem above.

**Comment by jimmybeckett** [comment]
Agreed @shahvi, good eye. My original plot did not exhibit the buggy behavior, but that appears to be due to environment. I was using the Anaconda interpreter with Pycharm. I saw the buggy behavior when I fired up JupyterLab. Is this the same environment you're using?

As a side note, you appear to be using Python 2.7. I recommend switching to Python 3 asap. The buggy behavior is present in both Python 2.7 and Python 3, but regardless switching is still a good idea.

**Comment by shahvi** [comment]
I have seen this issue with both Spyder and when running from command line. I do use Python 3 for most things, however some applications like ROS use 2.7, so forced to use it. And yes, I have seen the issue with both 2.7 and 3. I have been using this little script to overcome the marker issue, but would be good to fix it in the root.
```
def patch_legend(ax, *args, **kwargs):
    handles, labels = ax.get_legend_handles_labels()
    patches = []
    for handle, label in zip(handles, labels):
        patches.append(mpatches.Patch(color=handle.get_color(), label=label))
    legend = ax.legend(handles=patches, *args, **kwargs)
```

**Comment by jimmybeckett** [comment]
I did some more experimenting. The buggy behavior appears with any style that is not the default. More significantly, the buggy behavior does not appear when Python is ran from the command line. I'm tempted to write this off as a bug in conda instead of in Pandas.

**Comment by TomAugspurger** [comment]
More likely a difference in matplotlib backends, rather than a bug in conda.

So is it correct that we're still missing a reproducible example?

**Comment by jimmybeckett** [comment]
Yes, that is correct.

**Comment by Zomega** [comment]
I'll note that I'm experiencing this (or something very similar) on pandas 2.0.1, matplotlib 3.7.1, and Python 3.9.13, being run directly from the command line. I'm unfortunately not able to share the code for reproduction, but I will attach a cropped screenshot and see if I can boil it down to something for others to investigate. I am plotting multiple slices of the same dataframe onto the same ax.

![Screenshot (30)](https://github.com/pandas-dev/pandas/assets/879866/b5b390e1-e613-41da-9c50-28d6a0f9426c)
